lightning/src/ln/channelmanager.rs

```diff
@@ -2962,8 +2976,71 @@ where
 	/// [`Event::FundingGenerationReady::temporary_channel_id`]: events::Event::FundingGenerationReady::temporary_channel_id
 	/// [`Event::ChannelClosed::channel_id`]: events::Event::ChannelClosed::channel_id
 	pub fn create_channel(&self, their_network_key: PublicKey, channel_value_satoshis: u64, push_msat: u64, user_channel_id: u128, temporary_channel_id: Option<ChannelId>, override_config: Option<UserConfig>) -> Result<ChannelId, APIError> {
-		if channel_value_satoshis < 1000 {
-			return Err(APIError::APIMisuseError { err: format!("Channel value must be at least 1000 satoshis. It was {}", channel_value_satoshis) });
+		self.create_channel_internal(false, their_network_key, channel_value_satoshis, vec![], None,
+			push_msat, user_channel_id, temporary_channel_id, override_config)
+	}
+
+	/// Creates a new outbound dual-funded channel to the given remote node and with the given value
+	/// contributed by us.
+	///
+	/// `user_channel_id` will be provided back as in
+	/// [`Event::FundingGenerationReady::user_channel_id`] to allow tracking of which events
+	/// correspond with which `create_channel` call. Note that the `user_channel_id` defaults to a
+	/// randomized value for inbound channels. `user_channel_id` has no meaning inside of LDK, it
+	/// is simply copied to events and otherwise ignored.
+	///
+	/// `funding_satoshis` is the amount we are contributing to the channel.
```

Comment:
A concern arose while discussing #3558 : The logic here would be that the intended funding amount is specified, and the inputs that cover it. If the inputs constitute a larger amount, a change will also be added.
This comment here is in sync with that.
However, `funding_satoshis` is actually not an input parameter, but inferred from the inputs, with the assumption that the *entire* input amount is used for funding (and there is no change). Is that planned for later?